server/src/main/java/io/crate/replication/logical/MetadataTracker.java

```diff
@@ -460,23 +459,21 @@ static RestoreDiff getRestoreDiff(Subscription subscription,
         return new RestoreDiff(toRestoreIndices, toRestoreTemplates, relationNamesForStateUpdate);
     }
 
-    private ClusterState removeDroppedTablesOrPartitions(String subscriptionName,
-                                                         Subscription subscription,
-                                                         ClusterState subscriberClusterState,
-                                                         Response response) {
+    private ClusterState processDroppedTablesOrPartitions(String subscriptionName,
+                                                          Subscription subscription,
+                                                          ClusterState subscriberClusterState,
+                                                          Map<RelationName, RelationMetadata> relationsInPublications) {
         HashSet<RelationName> changedRelations = new HashSet<>();
-        HashSet<Index> indicesToRemove = new HashSet<>();
-        ArrayList<String> templatesToRemove = new ArrayList<>();
+        HashSet<Index> partitionsToRemove = new HashSet<>();
         Metadata subscriberMetadata = subscriberClusterState.metadata();
         for (var relationName : subscription.relations().keySet()) {
-            var publisherRelationMetadata = response.relationsInPublications().get(relationName);
-            var possibleTemplateName = PartitionName.templateName(relationName.schema(), relationName.name());
-            var isPartitioned = subscriberMetadata.templates().get(possibleTemplateName) != null;
-            boolean partitionDisappeared = isPartitioned && (publisherRelationMetadata == null || publisherRelationMetadata.template() == null);
```

Comment:
just to be sure - it looks like we don't use `publisherRelationMetadata.template()`anymore (and in general removed templates logic), so can drop it from `PublicationStateAction.Response`?